src/classes/GE_GiftEntryController.cls

```diff
@@ -65,16 +65,17 @@ public with sharing class GE_GiftEntryController {
             GE_PaymentServices.ResponseWrapper response = paymentServices.createTransaction();
 
             return JSON.serialize(response);
-        } catch (Exception ex) {
-            Boolean isCalloutException = ex instanceof System.CalloutException;
-            Boolean hasTimedout = isCalloutException && ex.getMessage().contains(TIMED_OUT);
+        } catch (CalloutException ex) {
+            Boolean hasTimedout = ex.getMessage().contains(TIMED_OUT);
 
             if (hasTimedout) {
                 return getPurchaseCallTimeoutResponse(ex, dataImportRecordId);
             } else {
-                String JSONExceptionData = ERR_ExceptionData.createExceptionWrapperJSONString(ex);
-                throw returnDmlException(JSONExceptionData);
+                return null;
             }
+        } catch(DmlException ex) {
```

Comment:
@voduyemi the issue there that I saw is that we are throwing an exception returned from `returnDmlException()` which only returns a `DmlException`. If we don't have any specific way we would handle other exceptions like a `JSONException`, for example, that exception would still throw on line 77 and display unhandled to the UI. It's only the specific exceptions where we've written methods to handle those types that I've added the more specific type for the catch parameter.

Reply:
I believe `returnDmlException` should not be used in this instance since the exception caught should be based on the action/call performed. This might not need happen in this WI to minimize the impact of your change since we specifically handle purchase call errors in a certain way to control what happens after a failed purchase attempt. The exception on line 77 will also not get thrown if a DMLException is not caught on line 76. If we still want to use `returnDmlException`, we can use the parent `Exception` class and to act like a 'catch-all' if other errors such as 
 the `JSONException` happens.

Reply:
Good point. I think the `AuraHandledException` is better to use as a catch-all so I could just replace `returnDmlException()` with `returnAuraException()` and use the `Exception` generic type. Main goal is to try and think about how to be more discrete about what errors we expect to occur and how we want to handle different exception types. Something I've been wanting to get better about. How does that sound?

Reply:
@dfuller114 That works, as long as we make sure that we are still able to handle the error thrown with `returnAuraException` gracefully in the front-end.

Reply:
@psadie-sfdo do you see any issue with changing this to an `AuraHandledException` instead of `DmlException`?

Reply:
Nope, I think that's definitely more appropriate than the `DmlException`

Reply:
I would just add to this thread that I would avoid using `GE_GiftEntryController.returnAuraException` or `GE_GiftEntryController.returnDmlException`.  I added a line (#15) to our refactor doc.  I think we should avoid using these and look to remove them.